src/c4/yml/emit.def.hpp

```diff
@@ -268,4 +268,172 @@ void Emitter<Writer>::_write_scalar(csubstr s)
 } // namespace yml
 } // namespace c4
 
+
+namespace c4 {
+namespace yml {
+
+template<class Writer>
+substr EmitterJSON<Writer>::emit(Tree const& t, size_t id, bool error_on_excess)
+{
+    this->_visit(t, id);
+    substr result = this->Writer::_get(error_on_excess);
+    return result;
+}
+
+template<class Writer>
+size_t EmitterJSON<Writer>::tell() const
+{
+    return this->Writer::m_pos;
+}
+
+template<class Writer>
+void EmitterJSON<Writer>::seek(size_t p)
+{
+    this->Writer::m_pos = p;
+}
+
+template<class Writer>
+size_t EmitterJSON<Writer>::_visit(Tree const& t, size_t id)
+{
+    if(t.is_stream(id))
+    {
+        ;
+    }
+    _do_visit(t, id);
+    if(t.is_stream(id))
+    {
+        _write("...\n");
+    }
+    return this->Writer::m_pos;
+}
+
+/** @todo this function is too complex. break it down into manageable pieces */
+template<class Writer>
+void EmitterJSON<Writer>::_do_visit(Tree const& t, size_t id)
+{
+  if(t.is_doc(id))
```

Comment:
watchout for indentation style

Reply:
I refactored the indentations in the entire document. Does it meet the projects style now?

Reply:
actually, no... compare to the original function.

Reply:
next attempt

Reply:
:-)

Almost there...! Logic blocks now are ok, but the whole code is indented with respect to the namespace. Deindent it one level, as this project's style does not indent the namespace.

Reply:
should be done now